resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -299,44 +272,78 @@ private[spark] class Client(
     })
   }
 
-  private def configurePorts(): (Seq[ServicePort], Seq[ContainerPort]) = {
-    val servicePorts = new ArrayBuffer[ServicePort]
-    val containerPorts = new ArrayBuffer[ContainerPort]
-
-    def addPortToServiceAndContainer(portName: String, portValue: Int): Unit = {
-      servicePorts += new ServicePortBuilder()
-        .withName(portName)
-        .withPort(portValue)
-        .withNewTargetPort(portValue)
-        .build()
-      containerPorts += new ContainerPortBuilder()
-        .withContainerPort(portValue)
-        .build()
+  private def getSubmitErrorMessage(
```

Comment:
👍 that `run()` method was getting ginormous -- I'd be happy breaking it up even more